src/steamship/agents/llms/steamship_llm.py

```diff
@@ -0,0 +1,110 @@
+        #  using specific PluginInstances rather than client.use_plugin all the time with the params that we determine?
+        self.client = plugin_instance.client
+        self.plugin_instance = plugin_instance
+
+    @staticmethod
+    def with_gpt4(client: Steamship, temperature: float = 0.4, max_tokens: int = 256):
+        # TODO (PR callout): This is an example of helpers that can initialize plugins that we know are solid / are
+        #  steamship-published.  These don't need to live in this class either, to keep the interface more pure.
+        gpt4 = client.use_plugin(
+            "gpt-4", config={"model": "gpt-4", "temperature": temperature, "max_tokens": max_tokens}
+        )
+        return SteamshipLLM(gpt4)
+
+    def generate(
+        self,
+        messages: List[Block],
+        system_prompt: List[Block] = None,
+        history: Optional[List[Block]] = None,
+        tools: Optional[List[Tool]] = None,
+        assert_capabilities: bool = True,
+        **kwargs,
+    ) -> List[Block]:
+        """
+        Call the LLM plugin's generate method.  Generate requests for plugin capabilities based on input parameters.
+
+        :param messages: Messages to be passed to the LLM to construct the prompt.
+        :param system_prompt: A system prompt to include with the prompt. Providing this requests the
+          SystemPromptSupport capability.
+        :param history: A list of blocks which are previous interactions with the tool. Providing this requests the
+          ConversationSupport capability.
+        :param tools: A list of Tools, representing functions which can be called with this LLM. Providing this requests
+          the FunctionCallingSupport capability, and provides the function definitions for it.
+        :param assert_capabilities: If True (default), raise a SteamshipError if the LLM plugin did not respond with a
+          block that asserts what level capabilities were fulfilled at.
+        :param kwargs: Options that can be passed to the LLM model as other parameters.
+        :return: a List of Blocks that are returned from the plugin.
+        """
+
+        # TODO (PR callout): I'm not certain this class needs to be abstract?  This seems pretty soup-to-nuts.
+        capabilities = []
+        blocks = messages[:]
+        if system_prompt:
+            blocks += system_prompt
+            capabilities.append(SystemPromptSupport())
+
+        if history:
+            blocks += history
+            capabilities.append(ConversationSupport())
+
+        if tools:
+            capabilities.append(
+                FunctionCallingSupport(functions=[tool.as_openai_function() for tool in tools])
+            )
+
+        blocks.append(CapabilityPluginRequest(requested_capabilities=capabilities).to_block())
+        # TODO (PR callout): I have maybe ideas for not needing temp files here in all cases? GH comment
```

Comment:
Probably outside the scope of this PR, but if we're already posting files, just post blocks to the client as part of the request eventually?